src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,249 @@
+package duke;
+
+import duke.task.Task;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * This class represents the User Interface, which is responsible for all the inputting and printing of
+ * statements which the user will see.
+ */
+public class Ui {
+    public static final String DEFAULT_SPACES = "    ";
+    public static final String LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private Scanner input;
+    public static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    /**
+     * Constructor for Ui. Initialises the input scanner.
+     */
+    public Ui() {
+        this.input = new Scanner(System.in);
+    }
+
+    /**
+     * Shows the welcome message.
+     * @return welcome message.
+     */
+    public static String showWelcome() {
+        return "Hello! I'm Duke\nWhat can I do for you?";
+    }
+
+    /**
+     * Gives the program the next line which was input by the user.
+     *
+     * @return string of the input.
+     */
+    public String readCommand() {
+        return input.nextLine();
+    }
+
+
+    /**
+     * Displays the tasks currently in the task list.
+     *
+     * @param taskList the current task list.
+     * @return the tasks in the list.
+     */
+    public static String listTasks(ArrayList<Task> taskList) {
+        if (taskList.size() == 0) {
+            return "You have no tasks in your list!\nTry adding some tasks with todo, event, or deadline.";
+        } else {
+            String output = "Here are the tasks in your list:\n";
+            for (int i = 0; i < taskList.size(); i++) {
+                output += (i + 1) + "." + taskList.get(i) + "\n";
+            }
+            return output;
+        }
+    }
+
+
+    /**
+     * Returns the String message instead of printing
+     *
+     * @param task task to be completed
+     * @return String of the message
+     */
+    public static String finishTask(Task task) {
+        String str = "Nice! I've marked this task as done:\n";
+        str = str + task;
+        return str;
+    }
+
+    /**
+     * Displays the remaining tasks in the list.
+     *
+     * @param tasks task list.
+     * @return the remaining tasks in the list.
+     */
+    public static String remainingTasks(ArrayList<Task> tasks) {
+        return String.format("Now you have %d tasks in the list.", tasks.size());
+    }
+
+    /**
+     * Displays a message when a task is deleted.
+     *
+     * @param task task to be deleted.
+     * @return message when task is deleted.
+     */
+    public static String deleteTask(Task task) {
```

Comment:
It might be helpful to clarify the method name to reflect that this method returns the delete task string rather than actually deleting the task?